Resources/doc/configuration_reference.md
CCDNForum ModeratorBundle Configuration Reference.
==================================================

All available configuration options are listed below with their default values.

``` yml
#
# for CCDNForum ModeratorBundle
#
ccdn_forum_moderator:
    user:
        profile_route: ccdn_user_profile_show_by_id
    template:
        engine: twig
	seo:
		title_length: 67
    flag:
        show_flagged:
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
            flags_per_page: 40
            topic_title_truncate: 50
            flag_created_datetime_format: 'd-m-Y - H:i'
            post_created_datetime_format: 'd-m-Y - H:i'
        show_flag:
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
        update_flag:
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
            form_theme: CCDNForumModeratorBundle:Form:fields.html.twig
    item_flag:
        flag_created_datetime_format: 'd-m-Y - H:i'
        moderated_datetime_format: 'd-m-Y - H:i'
    topic:
        change_board:
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
            form_theme: CCDNForumModeratorBundle:Form:fields.html.twig
        delete_topic: 
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
        show_closed: 
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
            topics_per_page: 40
            topic_title_truncate: 20
            post_created_datetime_format: 'd-m-Y - H:i'
            topic_closed_datetime_format: 'd-m-Y - H:i'
            topic_deleted_datetime_format: 'd-m-Y - H:i'
    post:
        show_locked:
            layout_template: CCDNComponentCommonBundle:Layout:layout_body_right.html.twig
            posts_per_page: 40
            topic_title_truncate: 20
            post_created_datetime_format: 'd-m-Y - H:i'
            post_locked_datetime_format: 'd-m-Y - H:i'
            post_deleted_datetime_format: 'd-m-Y - H:i'

```

- [Return back to the docs index](index.md).


DependencyInjection/Configuration.php
<?php

/*
 * This file is part of the CCDNForum ModeratorBundle
 *
 * (c) CCDN (c) CodeConsortium <http://www.codeconsortium.com/>
 *
 * Available on github <http://www.github.com/codeconsortium/>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace CCDNForum\ModeratorBundle\DependencyInjection;

use Symfony\Component\Config\Definition\ConfigurationInterface;
use Symfony\Component\Config\Definition\Builder\TreeBuilder;
use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;

/**
 *
 * This is the class that validates and merges configuration from your app/config files
 *
 * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html#cookbook-bundles-extension-config-class}
 *
 * @author Reece Fowell <reece@codeconsortium.com>
 * @version 1.0
 */
class Configuration implements ConfigurationInterface
{

    /**
     * {@inheritDoc}
     */
    public function getConfigTreeBuilder()
    {
        $treeBuilder = new TreeBuilder();
        $rootNode = $treeBuilder->root('ccdn_forum_moderator');

        // Here you should define the parameters that are allowed to
        // configure your bundle. See the documentation linked above for
        // more information on that topic.
        $rootNode
            ->children()
                ->arrayNode('user')
                    ->children()
                        ->scalarNode('profile_route')->defaultValue('ccdn_user_profile_show_by_id')->end()
                    ->end()
                ->end()
                ->arrayNode('template')
                    ->children()
                        ->scalarNode('engine')->defaultValue('twig')->end()
                    ->end()
                ->end()
            ->end();

        $this->addSEOSection($rootNode);
        $this->addFlagSection($rootNode);
        $this->addItemFlagSection($rootNode);
        $this->addTopicSection($rootNode);
        $this->addPostSection($rootNode);

        return $treeBuilder;
    }

    /**
     *
     * @access protected
     * @param ArrayNodeDefinition $node
     */
    protected function addSEOSection(ArrayNodeDefinition $node)
    {
        $node
            ->children()
                ->arrayNode('seo')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->scalarNode('title_length')->defaultValue('67')->end()
                    ->end()
                ->end()
            ->end();
    }

    /**
     *
     * @access private
     * @param ArrayNodeDefinition $node
     */
    private function addFlagSection(ArrayNodeDefinition $node)
    {
        $node
            ->children()
                ->arrayNode('flag')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->arrayNode('show_flagged')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                                ->scalarNode('flags_per_page')->defaultValue('40')->end()
                                ->scalarNode('topic_title_truncate')->defaultValue('50')->end()
                                ->scalarNode('flag_created_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                                ->scalarNode('post_created_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                            ->end()
                        ->end()
                        ->arrayNode('show_flag')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                            ->end()
                        ->end()
                        ->arrayNode('update_flag')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                                ->scalarNode('form_theme')->defaultValue('CCDNForumModeratorBundle:Form:fields.html.twig')->end()
                            ->end()
                        ->end()
                    ->end()
                ->end()
            ->end();
    }

    /**
     *
     * @access private
     * @param ArrayNodeDefinition $node
     */
    private function addItemFlagSection(ArrayNodeDefinition $node)
    {
        $node
            ->children()
                ->arrayNode('item_flag')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->scalarNode('flag_created_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                        ->scalarNode('moderated_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                    ->end()
                ->end()
            ->end();
    }

    /**
     *
     * @access private
     * @param ArrayNodeDefinition $node
     */
    private function addTopicSection(ArrayNodeDefinition $node)
    {
        $node
            ->children()
                ->arrayNode('topic')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->arrayNode('change_board')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                                ->scalarNode('form_theme')->defaultValue('CCDNForumModeratorBundle:Form:fields.html.twig')->end()
                            ->end()
                        ->end()
                        ->arrayNode('delete_topic')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                            ->end()
                        ->end()
                        ->arrayNode('show_closed')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                                ->scalarNode('topics_per_page')->defaultValue('40')->end()
                                ->scalarNode('topic_title_truncate')->defaultValue('20')->end()
                                ->scalarNode('post_created_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                                ->scalarNode('topic_closed_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                                ->scalarNode('topic_deleted_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                            ->end()
                        ->end()
                    ->end()
                ->end()
            ->end();
    }

    /**
     *
     * @access private
     * @param ArrayNodeDefinition $node
     */
    private function addPostSection(ArrayNodeDefinition $node)
    {
        $node
            ->children()
                ->arrayNode('post')
                    ->addDefaultsIfNotSet()
                    ->canBeUnset()
                    ->children()
                        ->arrayNode('show_locked')
                            ->addDefaultsIfNotSet()
                            ->children()
                                ->scalarNode('posts_per_page')->defaultValue('20')->end()
                                ->scalarNode('topic_title_truncate')->defaultValue('20')->end()
                                ->scalarNode('layout_template')->defaultValue('CCDNComponentCommonBundle:Layout:layout_body_right.html.twig')->end()
                                ->scalarNode('post_created_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                                ->scalarNode('post_locked_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                                ->scalarNode('post_deleted_datetime_format')->defaultValue('d-m-Y - H:i')->end()
                            ->end()
                        ->end()
                    ->end()
                ->end()
            ->end();
    }

}
